src/sentry/lang/javascript/processor.py

```diff
@@ -889,12 +898,17 @@ def cache_source(self, filename, release):
             cache.add_error(filename, exc.data)
             return
 
-        sourcemaps.add(sourcemap_url, sourcemap_idx)
+        sourcemaps.add(sourcemap_url, sourcemap_view)
 
         # cache any inlined sources
-        inline_sources = get_inline_content_sources(sourcemap_idx, sourcemap_url)
-        for source in inline_sources:
-            self.cache.add(*source)
+        # inline_sources = sourcemap_view.get_inline_content_sources(sourcemap_url)
+        for src_id, source in sourcemap_view.iter_sources():
+            # TODO(mattrobenolt): This is slightly less than ideal,
+            # but it's the simplest path for now.
+            # Ideally, we would do this lazily.
+            content = sourcemap_view.get_source_contents(src_id)
```

Comment:
for the libsourcemap version this should not be done as fetching sourcecode from it is slow (particularly from memdb). Ideally we restructure the cache so that it only knows about how to fetch source but to not actually do it. (eg: register src_id + sourcemap) with the filename.